src/main/java/com/sgio/yieldseeker/builder/PurchaseBuilder.java

```diff
@@ -0,0 +1,86 @@
+package com.sgio.yieldseeker.builder;
+
+import com.google.gson.JsonObject;
+import com.sgio.yieldseeker.model.Apartment;
+import com.sgio.yieldseeker.model.Purchase;
+import com.sgio.yieldseeker.service.CalculatorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PurchaseBuilder {
+
+    @Autowired
+    private CalculatorService calculatorService;
```

Comment:
```suggestion
@Component
@RequiredArgsConstructor
public class PurchaseBuilder {
    private final CalculatorService calculatorService;
```
Avoid autowired, use constructor-based dependency injection instead